lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -1657,6 +1658,7 @@ Instruction *InstCombiner::visitICmpInstWithInstAndIntCst(ICmpInst &ICI,
                           Builder->CreateTrunc(LHSI->getOperand(0), NTy),
```

Comment:
You might be interested to know that the latest llvm code adds ```DL.isLegalInteger(TypeBits-Amt)``` to this conditional. Assuming we have ```LegalIntWidths``` set up properly, I expect that might allow us to occasionally benefit from this code while avoiding invalid integer types. Then again, I know that 16 bit types are not very popular.

Reply:
https://github.com/llvm/llvm-project/blob/7f5fe30a150e7e87d3fbe4da4ab0e76ec38b40b9/llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp#L2148

Reply:
We don't want anything changing the int type being used based on this.  Our DataLayout will say we support 8, 16, 32, and 64 bit types, but 8-bit is only supported for certain constants on intrinsics, and 16 or 64 bit introduces a shader feature requirements that shouldn't be introduced by optimizations if the user didn't want them.  This optimization is still easy enough to perform by the DXIL consumer, since we aren't targeting hardware instructions here.  I don't see any reason to do it for DXIL.

Reply:
Sounds reasonable!